lib/tweet.rb

```diff
@@ -0,0 +1,30 @@
+class Author
+  attr_accessor :name
+  attr_accessor :email
+
+  def initialize(name: nil, email: nil)
+    @name = name
+    @email = email
+  end
+end
+
+class Tweet
+  attr_reader :author, :message, :id
+
+  def initialize(author:, message:, id:)
+    @author = author
+    @message = message
+    @id = id
+  end
+
+  def self.build_from_hash(tweet)
+    Tweet.new(
+      author: Author.new(
+        name: tweet['author']['name'],
+        email: tweet['author']['email']
+      ),
```

Comment:
This is very simple (in this case), but very important case.

Right now the `Tweet` class has knowledge about properties of author (name, email). If you would add `surename` to the author, you **would have to change two classes - `Tweet` and `Author`**. This is not very good, right?

Reply:
The good news is that you can fix it easily (and you already know how, because you have already done it with `Tweet` ;)):

```suggestion
      author: Author.build_from_hash(tweet['author']),
```